go/ssl/ssl.go

```diff
@@ -54,19 +54,30 @@ func NewTLSConfig(caFile string, verifyCert bool) (*tls.Config, error) {
 		log.Info("verifyCert requested, client certificates will be verified")
 		c.ClientAuth = tls.VerifyClientCertIfGiven
 	}
+	caPool, err := ReadCAFile(caFile)
+	if err != nil {
+		return &c, err
+	}
+	c.ClientCAs = caPool
+	c.BuildNameToCertificate()
```

Comment:
I wonder if this enforces an Authority, or will this still allow self-signed certificates?

Reply:
We use self-signed certs with a local root CA certificate without any problems.
Actually problem was not in this line. This isn't my code, i just reused old version of code for reading CA certificate in new `ReadCAFile` function.
This function is used in client and server now because for server you need `ClientCAs` and `RootCAs` for client.